Description: Make the script that fetches the Node.js release tarball architecture-aware
Author: Olivier Tilloy <olivier.tilloy@canonical.com>
Forwarded: not-needed

Index: chromium-135.0.7049.52/third_party/node/update_node_binaries
===================================================================
--- chromium-135.0.7049.52.orig/third_party/node/update_node_binaries
+++ chromium-135.0.7049.52/third_party/node/update_node_binaries
@@ -138,10 +138,15 @@ if [[ $upload != true ]]; then
   # First download checksum file.
   rm -f "SHASUMS256.txt"
   wget "https://nodejs.org/dist/${NODE_VERSION}/SHASUMS256.txt"
-  download_unix "darwin-x64" "mac"
-  download_unix "darwin-arm64" "mac"
-  download_unix "linux-x64" "linux"
-  download_win
+  if [ $(arch) = "x86_64" ]; then
+    download_unix "linux-x64" "linux"
+  elif [ $(arch) = "armv7l" ]; then
+    download_unix "linux-armv7l" "linux"
+    mv linux/node-linux-armv7l linux/node-linux-x64
+  elif [ $(arch) = "aarch64" ]; then
+    download_unix "linux-arm64" "linux"
+    mv linux/node-linux-arm64 linux/node-linux-x64
+  fi
   echo "To upload the binaries, please run ./update_node_binaries --upload"
 else
   echo "Uploading to GCS and updating DEPS files..."
